src/app/modules/spcp/spcp.util.ts

```diff
@@ -1,13 +1,40 @@
 import { StatusCodes } from 'http-status-codes'
 
 import { createLoggerWithLabel } from '../../../config/logger'
-import { MapRouteError } from '../../../types'
+import { AuthType, MapRouteError, MyInfoAttribute } from '../../../types'
 import { MissingFeatureError } from '../core/core.errors'
 
 import { CreateRedirectUrlError } from './spcp.errors'
 
 const logger = createLoggerWithLabel(module)
 
+type SpcpLocals =
+  | {
+      uinFin: string
+      hashedFields: Set<MyInfoAttribute>
+    }
+  | { uinFin: string; userInfo: string }
+  | { [key: string]: never } // empty object
+export const getMockSpcpLocals = (
```

Comment:
I think this should be in a service, since it is production functionality after all

Reply:
okay! I think it belongs in the AdminForm service since it's more of admin functionality than SPCP functionality, so I'll move it there